### PR TITLE
handle loss of internet connection when submitting forms in webapps

### PR DESCRIPTION
**Pull request by biyeun**
Addresses the issues described in [HI-562](https://dimagi-dev.atlassian.net/browse/HI-652)

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js [hunk 1]
```diff
@@ -641,6 +641,8 @@ Formplayer.Errors = {
         "Internet in order to submit your form.",
     LOCK_TIMEOUT_ERROR: gettext('Another process prevented us from servicing your request. ' +
         'Please try again later.'),
+    NO_INTERNET_ERROR: gettext("We have detected an issue with your network. " +
+        "Please check your Internet connection"),
```
Comment (orangejenny) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js]: Do you think it's worth explicitly telling them they can retry once they have connectivity?

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js (hunk 1 above)
Comment (biyeun) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js]: yeah. that might be worth it.

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/webformsession.js [hunk 2]
```diff
@@ -243,7 +243,15 @@ WebFormSession.prototype.handleFailure = function (resp, action, textStatus, fai
     } else if (textStatus === 'timeout') {
         errorMessage = Formplayer.Errors.TIMEOUT_ERROR;
     } else if (resp.hasOwnProperty('responseJSON')) {
-        errorMessage = Formplayer.Utils.touchformsError(resp.responseJSON.message);
+        if (resp.responseJSON === undefined) {
```
Comment (esoergel) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/webformsession.js]: Is there a more direct way to check for a connectivity issue?  Like some particular response code or checking `window.navigator.onLine`?

Comment (biyeun) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/webformsession.js]: updated

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js [hunk 3]
```diff
@@ -302,6 +302,22 @@ FormplayerFrontend.on("start", function (options) {
...
+            $('.form-control').prop('disabled', 'disabled');
+        }
+    );
+    window.addEventListener(
+        'online', function () {
+            showSuccess(gettext("You are are back online."), $("#cloudcare-notifications"));
+            $('.submit').removeAttr('disabled');
+            $('.form-control').removeAttr('disabled');
+        }
+    );
```
Comment (esoergel) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js]: This is neat

**Review by esoergel — APPROVED** [review]
Thanks!

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js (hunk 3 above)
Comment (orangejenny) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js]: slick!

**Review by kaapstorm — APPROVED** [review]

**Comment by kaapstorm** [comment]
:recycle: :construction_worker_woman:

**Pull request merged**